libs/ec/inc/public/tfc/devices/abt/easycat.hpp

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <bitset>
+#include <cstdio>
+
+#include "ethercat.h"
+#include "tfc/devices/base.hpp"
+
+#include "tfc/ipc.hpp"
+
+namespace tfc::ec::devices::abt {
+class easyecat : public base {
+public:
+  explicit easyecat(boost::asio::io_context& ctx) : base(ctx) {}
+  static constexpr auto vendor_id = 0x79a;
+  static constexpr auto product_code = 0xdefede;
+  auto process_data(uint8_t* input, uint8_t* output) noexcept -> void final {
+    if (input == nullptr) {
```

Comment:
maybe add `|| output == nullptr` so we won't accidentally deref nullptr below

Reply:
I could do this check above ( where process_data is called ) but taking the parameter in as a reference is confusing
as this is not a "uint8_t" this is just a pointer to a buffer of memory. 

I would have prefered to use something that verifies the size of the input and does not have the option of being a nullptr. This function signature is something I would like to change.